pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -150,6 +152,83 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		}
 	*/
 
+	networkAddresses := []apicorev1.NodeAddress{}
+
+	// The computer name for a VM instance is the hostname of the VM
+	// TODO(jchaloup): find a way how to propaget the hostname change in case
+	// someone/something changes the hostname inside the VM
+	if vm.OsProfile != nil && vm.OsProfile.ComputerName != nil {
+		networkAddresses = append(networkAddresses, apicorev1.NodeAddress{
+			Type:    apicorev1.NodeHostName,
+			Address: *vm.OsProfile.ComputerName,
+		})
+	}
+
+	if vm.NetworkProfile != nil && vm.NetworkProfile.NetworkInterfaces != nil {
+		for _, iface := range *vm.NetworkProfile.NetworkInterfaces {
+			// Get iface name from the ID
+			ifaceName := path.Base(*iface.ID)
+			networkIface, err := s.networkInterfacesSvc.Get(ctx, &networkinterfaces.Spec{
+				Name:     ifaceName,
+				VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
+			})
+			if err != nil {
+				klog.Errorf("Unable to get %q network interface: %v", ifaceName, err)
+				continue
+			}
+
+			niface, ok := networkIface.(network.Interface)
+			if !ok {
+				klog.Errorf("Unable to cast %q to network.Interface: %v", ifaceName, err)
```

Comment:
Report what the actual type turned out to be (via `%T`). Would help debugging if we run into this.

Reply:
Also logging `err` here seems wrong.

Reply:
And the message contains `ifaceName` which also seems wrong; suspecting copy/paste here :)

Reply:
`ifaceName` is intended, it's name of the interface